pkg/crd/gen.go

```diff
@@ -39,6 +39,12 @@ type Generator struct {
 	// Kubernetes API servers.  The storage version's schema will be used as
 	// the CRD's schema.
 	TrivialVersions bool `marker:",optional"`
+
+	// MaxDescLen limits the description length of each field in CRD's OpenAPI schema.
+	//
+	// 0 value (default) indicates no limit on description of fields.
+	// -1 value indicates drop the description for all fields.
+	MaxDescLen int `marker:",optional"`
```

Comment:
IMO it'd better to change it to a pointer:
- nil means no limit (default)
- 0 means drop description completely
- n means at most n characters

@DirectXMan12 Pointer should work here, right? If not, we probably should fix it.

Reply:
agree with @mengqiy here

Reply:
Good suggestion, refactored it to take ptr.